include/openssl/ssl3.h

```diff
@@ -357,7 +357,7 @@ extern "C" {
  * SSL3_CT_NUMBER is used to size arrays and it must be large enough to
  * contain all of the cert types defined either for SSLv3 and TLSv1.
  */
-# define SSL3_CT_NUMBER                  9
+# define SSL3_CT_NUMBER                  10
```

Comment:
I've been trying to figure this number out because it just doesn't make sense. Before your additions there are 7 certificates types defined here plus an additional 4 in tls1.h (excluding repeats) - so that means the number should have been 11 not 9.

So I did a bit more digging. It turns out that the comment above this definition is completely misleading. The value SSL3_CT_NUMBER is used to size the tmp.ctype array in the SSL structure. This is used to record the certificate types that a server is willing to accept when a client receives a CertificateRequest message. Historically that was the only place we stored the acceptable certificate types - so if we received more than would fit in the array then we would have a problem. Therefore it would be more accurate if the comment actually said:
"...it must be large enough to contain all of the cert types the are currently defined or will ever be defined in any current or future RFC".

In other words it has nothing to do with the number of cert types that OpenSSL itself knows about. That is of course completely ridiculous and unworkable, so some time ago Steve implemented a "fix" which basically means that if we receive more certificate types than will fit into the array then we allocate some dynamic memory and store them there instead.

The result of all of that is I think we should leave SSL3_CT_NUMBER and TLS_CT_NUMBER (which isn't used *anywhere*) alone. It looks like there needs to be some separate work to clean it up - but that's not for you to worry about in this patch.